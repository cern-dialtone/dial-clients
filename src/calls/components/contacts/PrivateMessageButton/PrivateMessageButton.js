import React from 'react';
import { Button } from 'semantic-ui-react';

export default function PrivateMessageButton({ profile }) {
  return (
    <a
      href={
        'https://mattermost.web.cern.ch/_redirect/messages/@' + profile.username
      }
      target="_blank"
      rel="noopener noreferrer"
    >
      <Button fluid className="CalleeProfileNumber" role="button">
        <i aria-hidden="true" className="chat icon"></i>
        Send private message
      </Button>
    </a>
  );
}
