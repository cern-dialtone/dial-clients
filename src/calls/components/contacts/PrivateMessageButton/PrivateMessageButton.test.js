import React from 'react';
import { render } from '@testing-library/react';
import PrivateMessageButton from './PrivateMessageButton';

describe('ContactProfileModal tests', () => {
  it('renders the modal closed', async () => {
    const { asFragment, queryByTestId } = render(
      <PrivateMessageButton
        profile={{
          personId: '4',
          mail: 'testmail@cern.ch',
          physicalDeliveryOfficeName: 'Name',
          username: 'username',
          phones: [],
          displayName: 'User Name',
          division: 'IT',
          cernGroup: 'CDA',
          cernSecction: 'IC'
        }}
      />
    );
    const modal = queryByTestId('PrivateMessageButton');
    expect(modal).toBeNull();

    expect(asFragment()).toMatchSnapshot();
  });
});
