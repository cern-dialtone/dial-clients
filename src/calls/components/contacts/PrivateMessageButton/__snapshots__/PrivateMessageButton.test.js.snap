// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactProfileModal tests renders the modal closed 1`] = `
<DocumentFragment>
  <a
    href="https://mattermost.web.cern.ch/_redirect/messages/@username"
    rel="noopener noreferrer"
    target="_blank"
  >
    <button
      class="ui fluid button CalleeProfileNumber"
      role="button"
    >
      <i
        aria-hidden="true"
        class="chat icon"
      />
      Send private message
    </button>
  </a>
</DocumentFragment>
`;
